internal/handler/userHandler.go

```diff
@@ -56,17 +58,27 @@ func (h *Handler) Save(context *gin.Context) {
 	if err != nil {
 		logger.Error("Error while creating user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while mapping request to user: " + err.Error(),
+			Message:   "Error while mapping request to user",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
 		return
 	}
 	createdUser, err := h.service.SaveUser(context, user)
 	if err != nil {
+		var databaseConnectionErr *exceptions.DatabaseConnectionErr
+		if errors.As(err, &databaseConnectionErr) {
+			logger.Errorf("Error while connecting to database. %v", err)
+			errorResponse := response.ErrorDto{
+				Message:   "Something went wrong",
+				Timestamp: time.Now(),
+			}
+			context.IndentedJSON(http.StatusInternalServerError, errorResponse)
+			return
+		}
 		logger.Error("Error while persisting user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while creating user. " + err.Error(),
+			Message:   "Error while creating user.",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
```

Comment:
There's no need to check for a database specific error. If the save call returned an error there's no way we can recover so we should log the error and return a 500 status with a generic error message.

Reply:
But my method returns an specific error for databaseConnection and for userNotFound, for example, I need to check which error im receiving, so I can return the correct status code and message

Reply:
I don't know if this is a best practice, since it's an error from user-side, we should return an error. It will be more easier to understand what is happening, given that  we will threat with something like this: if user == nil {logError}
The difference is just that I'm getting an error instead of verifying user nil.
But as you mentioned that is what we do here, I can change it.